patches/server/0872-Added-and-implemented-PlayerLootDropEvent.patch

```diff
@@ -0,0 +1,72 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NeonOverflow <nneonoverflow@protonmail.com>
+Date: Thu, 13 Jan 2022 22:58:28 -0500
+Subject: [PATCH] Added and implemented PlayerLootDropEvent.
+
+As a result of how the PlayerDeathEvent was implemented, when a player dies their items are simply dropped
+through the "world.dropItem()" method. This makes it very annoying for plugin developers to ultimately
+track down the Item entities dropped when a player dies. This patch implements a "PlayerLootDropEvent",
+making it easy for developers to track down the Item entities dropped when a player dies. Furthermore,
+Item entities dropped on player death are marked as being thrown by the player that died.
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 35cc150adf51f79e2fccef8b094c90554aafbee2..c84de6cb47ee8335ced754dacdeb27e388551f3b 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -907,15 +907,34 @@ public class CraftEventFactory {
+         victim.expToDrop = event.getDroppedExp();
+         victim.newExp = event.getNewExp();
+ 
++        // Paper start - implement player loot drop event
++        List<Item> items = new ArrayList<Item>();
++
+         for (org.bukkit.inventory.ItemStack stack : event.getDrops()) {
+             if (stack == null || stack.getType() == Material.AIR) continue;
+ 
+-            world.dropItem(entity.getLocation(), stack);
++            Item item = (Item) victim.drop(CraftItemStack.toNMSStack(stack), true, false);
```

Comment:
Did you actually test this? You can't just cast an nms ItemEntity to the bukkit Item interface.

Reply:
That was.... a pretty dumb mistake. If I had to guess I probably didn't build probably and ended up using an earlier version before that change was made when I tried testing it. My bad.